### ISSUE TITLE
Placement (native path): capture-compat is YES-as-seam (~tens of us); but doc's embedding-gather target is resident on qwen14b-zp (streamed 389MB/token is lm_head)

### ISSUE THREAD

**Issue by justinchuby** [justinchuby/onnx-genai#1299]
## Summary

The CUDA-graph-capture blocker on **placement** (native path) named in `docs/memory/MEMORY_MANAGEMENT_MODEL_DESIGN.md` and tracked in #994/#854/#867 is now resolved empirically, and — separately — the doc's *worked example* (host-place the low-arithmetic-intensity embedding gather to avoid streaming a 389 MB table per token) **does not apply to the current `qwen14b-zp` export**. Details below. Measured on an RTX 4060 (8 GB, driver 591.55, CUDA 13.1 runtime, WDDM), `--features cuda --lib -- --test-threads=1`.

## Finding 1 — capture compatibility: YES, as an eager seam (native path only)

A per-token device excursion (device→host→device) **is** compatible with native CUDA graph capture, **only** when treated as an eager seam between captured segments — which the native segmented-capture machinery already does. It is **illegal inside** an active capture: a host-consuming D2H needs a stream synchronize, which invalidates an in-flight capture. Both arms are now locked in by a permanent test (`host_excursion_is_capturable_as_a_seam_and_illegal_inside_capture`, merged in #1297).

**Price of admission** (permanent bench `bench_host_seam_price_of_admission`, #1297): splitting a monolithic captured step into two segments around one host seam costs **~47–90 µs/token** on the RTX 4060, with the segmented+seam output asserted **bit-identical** to the monolithic reference. Orders of magnitude below a bandwidth-bound weight stream.

This closes the "unknown — could invalidate placement on the native path" question in the affirmative for the **native** path. **Says nothing about the plugin-EP interspersed-partition hang (#982), which is untouched.**

## Finding 2 — the doc's placement *target* does not exist in this export

The doc's worked example assumes the **embedding gather** streams a 389 MB table per token (low arithmetic intensity: 389 MB in, ~10 KB out) and would benefit from host placement. Empirically, on `qwen14b-zp`:

- The 389 MB tensor that **is** streamed every token (weight-paging key **919**, `reads_per_step=1.000, hits=0, bypass_page_ins=40`) is **`lm_head.weight`**, consumed by **`MatMulNBits`** — the vocab projection (152064 × 5120 × 0.5 B INT4 = 389,283,840 B exactly). Verified by dumping lazy-weight handles: `value#919 name="lm_head.weight" boundary=MatMulNBits`.
- The **embedding gather** `model.embed_tokens.qweight` is consumed by **`GatherBlockQuantized`**, which is **not** a `LazyWeightBoundary`, so it is **never paged** — it is **resident** and streams **nothing** per token. Confirmed: of 867 lazy (paged) weights, **zero** are named `embed_tokens`.

So host-placing the embedding gather would save nothing (it is already resident), and the tensor that *does* stream (lm_head) is a real matmul producing the full vocab logits, not a low-AI gather.

## Consequence / open question

- The greenlit "host-place the embedding gather" work has **no target** on this export and was **not built**.
- `lm_head.weight` (MatMulNBits, streamed 389 MB + 48.6 MB scales/token) is itself **memory-bound / low-ish arithmetic intensity** and *might* be a placement/compute-on-host candidate (host RAM BW ~5× PCIe), but that is a **different op** requiring a host INT4 GEMV kernel and its own go/no-go — not the doc's example. Filing for a decision, not building.

## Reproduce

```
$env:ONNX_GENAI_WEIGHT_PAGING_KEY_TRACE=1; $env:ONNX_GENAI_WEIGHT_OFFLOAD=1
profile_native --model <qwen14b-zp> --ep cuda --backend native --tokens 40 --steady --decode-skip 8
```
`weight_paging_key: key=919 len=389283840 reads_per_step=1.000 hits=0 bypass_page_ins=40` is `lm_head.weight`; no `embed_tokens` key appears.

## Doc update needed

`MEMORY_MANAGEMENT_MODEL_DESIGN.md` should note that (a) native-path capture is compatible with a per-token host seam (~tens of µs), and (b) on quantized qwen exports the embedding gather is resident (`GatherBlockQuantized`, not a lazy boundary) — the streamed 389 MB/token is the `lm_head` projection, so the embedding-gather placement example does not apply as written.
